packages/neon-api/src/provider/neoscan/core.ts

```diff
@@ -120,10 +122,14 @@ export async function getBalance(
     if (b.amount > 0 && b.unspent.length > 0) {
       bal.addAsset(b.asset, {
         unspent: parseUnspent(b.unspent),
+        hash: b.asset_hash,
       } as Partial<wallet.AssetBalanceLike>);
     } else {
       bal.addToken(b.asset, b.amount);
     }
+    if (b.asset && b.asset_hash) {

```

Comment:
This means that we are adding a token hash even if the logic is adding an asset instead. I do not think that is the correct case.

Reply:
correction applied